### PR TITLE
Update packages to be browser-first and use separate export for Node.js

### PR DESCRIPTION
**Pull request by Mrtenz**
This refactors all packages to be browser-first: All APIs and packages used should be compatible with browsers. This means no use of Node.js builtins, or native packages. Node-specific APIs have been moved to a separate export `/node` (e.g., `@metamask/snaps-controllers/node`). For the `snaps-controllers` package I've also added a `/react-native` export, which exports the React Native webview service.

This also means we can remove the "browser" field from the packages that were using it.

## Breaking changes

- Anything that uses Node.js was removed from the default export, and needs to be imported from `/node`.
- The React Native webview service was moved, and needs to be imported from `/react-native`.

### CODE REVIEW

File: .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch
Comment (Mrtenz) [inline .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch:10]: I'm not sure if this is a bug or not, but the previous behaviour was unexpected (and I'm not sure how this worked before). Basically when checking a `tsconfig.json` path with a wildcard, this function returns the replacement for the path.

- Pattern: `@metamask/*/node`
  Input: `@metamask/snaps-utils/node`
  Result: `snaps-utils`
- Pattern: `@metamask/*`
  Input: `@metamask/snaps-controllers`
  Result: `snaps-controllers`

And so on. Before this change however, it would return:

- Pattern: `@metamask/*/node`
  Input: `@metamask/snaps-utils/node`
  Result: `snaps-utils/node`

Meaning that it would resolve to `packages/snaps-utils/node/src/...` and some other invalid paths. This causes it to fall back to Node.js' default resolver, which would look at the `package.json` of `snaps-utils` and try to load `packages/snaps-utils/dist/node.js`, but we can't do that since the file may not be built yet.

File: constraints.pro
Comment (FrederikBolding) [inline constraints.pro:208]: Why are we making this change?

File: packages/snaps-utils/src/node.ts
Comment (FrederikBolding) [inline packages/snaps-utils/src/node.ts:1]: Why do we need to disable a lint rule here?

File: constraints.pro
Comment (Mrtenz) [reply constraints.pro:208]: I changed the build command in `snaps-execution-environments` to build the LavaMoat packages too.

File: packages/snaps-utils/src/node.ts
Comment (Mrtenz) [reply packages/snaps-utils/src/node.ts:1]: There's some duplicate exports because we're re-exporting `./manifest/node` and `./virtual-file/node`, and parts of that are also re-exported in `index.ts`.

File: .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch
Comment (FrederikBolding) [inline .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch:10]: We need to try and upstream this. Also we need to determine if it is a bug before merging 😓

File: packages/snaps-utils/src/node.ts
Comment (FrederikBolding) [inline packages/snaps-utils/src/node.ts:3]: This is a nice pattern, we should let everyone in the team know about how exports will work now.

Comment (Mrtenz) [reply packages/snaps-utils/src/node.ts:3]: That was my plan after merging this yeah.

File: packages/snaps-utils/src/node.ts
Comment (Mrtenz) [reply packages/snaps-utils/src/node.ts:1]: We have [an eslint-config for browser environments](https://github.com/MetaMask/eslint-config/tree/main/packages/browser), but it would be quite tedious to only enable this for certain files. Do you have any good ideas for how to do this?

File: .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch
Comment (FrederikBolding) [reply .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch:10]: Can we try opening a well-reasoned PR with tests etc that is an easy merge for the maintainers?

File: constraints.pro
Comment (Mrtenz) [reply constraints.pro:208]: Good thing Yarn 4 supports JavaScript-based constraints. 😅

File: .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch
Comment (Mrtenz) [reply .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch:10]: Sure, but do you think that should block merging this?

File: packages/snaps-utils/src/node.ts
Comment (FrederikBolding) [reply packages/snaps-utils/src/node.ts:1]: Not really, that's why I ask 😅 

I guess we could see if we can reduce this list and use that build failing as heuristic? https://github.com/MetaMask/snaps/blob/main/packages/snaps-simulator/webpack.config.ts#L63-L81

But it's not great.

File: .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch
Comment (FrederikBolding) [reply .yarn/patches/tsconfig-paths-npm-3.14.2-90ce75420d.patch:10]: It would be nice to get confirmation that it's a bug, but probably not blocking

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**
